framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +86,99 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
+			height: aggregateCommit.height,
+		});
+
+		const certificate = computeCertificateFromBlockHeader(block.header);
+		certificate.aggregationBits = block.header.aggregateCommit.aggregationBits;
+		certificate.signature = aggregateCommit.certificateSignature;
+
+		return certificate;
+	}
+
+	public async getNextCertificateFromAggregateCommits(
+		lastCertifiedHeight: number,
+		aggregateCommits: AggregateCommit[],
+	): Promise<Certificate | null> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
+			height: lastCertifiedHeight,
+		});
+
+		const validatorsData = this._chainConnectorDB.getChainConnectorInfo().validatorsHashPreimage[
+			block.header.validatorsHash!.toString('hex')
+		];
+
+		const blsKeyToBFTWeight: Record<string, BigInt> = {};
+
+		validatorsData.validators.forEach((validator: BFTValidator) => {
+			blsKeyToBFTWeight[validator.blsKey.toString('hex')] = validator.bftWeight;
+		});
+
+		const bftHeights = await this._mainchainAPIClient.invoke<BFTHeights>('consensus_getBFTHeights');
+
+		let height = bftHeights.maxHeightCertified;
+
+		while (height > lastCertifiedHeight) {
+			if (aggregateCommits[height] !== undefined) {
```

Comment:
I'm not sure, if this `aggregateCommits[height]` translates correctly to what is specified in the LIP0061, [LIP0061 Reference](https://github.com/LiskHQ/lips/blob/536b2e6e0eb7670127dd955c2d1ecfa6c11c99cd/proposals/lip-0061.md?plain=1#L635).